pos_require_customer/static/src/js/models.js

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2020 Akretion (https://www.akretion.com)
+ * @author Raphaël Reverdy <raphael.reverdy@akretion.com>
+ * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). */
+
+odoo.define("pos_require_customer.models", function (require) {
+    "use strict";
+
+    var models = require("point_of_sale.models");
+
+    models.PosModel.prototype.models.some(function (model) {
+        if (model.model !== "product.pricelist") {
+            return false;
+        }
+        model.fields.push("pos_require_customer");
+        return true; // Exit early the iteration of this.models
+    });
```

Comment:
Use `load_fields` instead:

https://github.com/odoo/odoo/blob/c0295d5844e2924e684312245e56e82be92458c1/addons/point_of_sale/static/src/js/models.js#L1435-L1439


```suggestion
    models.load_fields("product.pricelist", ["pos_require_customer"]);
```